lib/pbench/server/api/resources/datasets_list.py

```diff
@@ -0,0 +1,151 @@
+
+    def _get(self, json_data: JSON, request: Request) -> Response:
+        """
+        Get a list of datasets matching a set of criteria.
+
+        NOTE: This does not rely on a JSON payload to identify the dataset and
+        desired metadata keys; instead we rely on URI query parameters.
+
+        TODO: The dispatcher is currently merging JSON request body and Flask
+        URI keywords ... it probably makes sense to also merge request query
+        parameters and then validate them all together; although it might make
+        more sense to validate each against separate schemas.
+
+        GET /api/v1/datasets/list?start=1970-01-01&end=2040-12-31&owner=fred&metadata=dashboard.seen&metadata=server.deletion
+        """
+
+        # We missed automatic schema validation due to the lack of a JSON body;
+        # construct an equivalent JSON body now so we can run it through the
+        # validator.
+        json = {}
+        for key in request.args.keys():
+            if key in self.schema:
+                if self.schema[key].type == ParamType.LIST:
+                    json[key] = request.args.getlist(key)
+                else:
+                    json[key] = request.args.get(key)
+
+        # Normalize and validate the metadata keys we got via the HTTP query
+        # string. These aren't automatically validated by the superclass, so we
+        # have to do it here.
+        try:
+            self.logger.info("pre-validate {!r}", json)
+            new_json = self.schema.validate(json)
+            self.logger.info("post-validate {!r}", new_json)
+        except SchemaError as e:
+            self.logger.warning("Schema validation error {}", e)
+            abort(HTTPStatus.BAD_REQUEST, message=str(e))
+
+        # Validate the authenticated user's authorization for the combination
+        # of "owner" and "access".
+        self._check_authorization(json_data.get("owner"), new_json.get("access"))
+
+        # Build a SQLAlchemy Query object expressing all of our constraints
+        #
+        # TODO: Think about what other constraints we might want to support.
+        # Including Metadata outside the Dataset table would be more difficult,
+        # but should be possible.
+        query = Database.db_session.query(Dataset)
+        if "start" in new_json and "end" in new_json:
+            self.logger.info("Adding start / end query")
+            query = query.filter(
+                Dataset.created.between(new_json["start"], new_json["end"])
+            )
+        elif "start" in new_json:
+            self.logger.info("Adding start query")
+            query = query.filter(Dataset.created >= new_json["start"])
+        elif "end" in new_json:
+            self.logger.info("Adding end query")
+            query = query.filter(Dataset.created <= new_json["end"])
+        if "name" in new_json:
+            self.logger.info("Adding name query")
+            query = query.filter(Dataset.name.contains(new_json["name"]))
```

Comment:
Yeah, well, if you already know the name you want, there's no point in using `datasets/list`, because it's not a list; use `datasets/detail` instead. (Which you'll probably want to do with the results of `datasets/list` anyway.)

And I did very briefly consider start or end or even a full regex, and decided it just wasn't worthwhile right now. We can always do that later.